ui/src/DataService.js

```diff
@@ -0,0 +1,46 @@
+const getWSUrl = () => {
+  const loc = window.location;
+  let newUrl;
+  if (loc.protocol === 'https:') {
+    newUrl = 'wss:';
+  } else {
+    newUrl = 'ws:';
+  }
+  newUrl += `//${loc.host}`;
+  return newUrl;
+};
+
+const baseUrl = '/api';
+const wsUrl = `${getWSUrl()}/api`;
+
+export const wsError = {};
+
+const requestConfig = (method, body) => ({
+  method,
+  cache: 'no-cache',
+  credentials: 'same-origin',
+  headers: {
+    'Content-Type': 'application/json; charset=utf-8'
+  },
+  body: body && JSON.stringify(body)
+});
+
+const request = async (url, method, body) => {
+  const response = await fetch(`http://localhost:3000${baseUrl}/${url}`, requestConfig(method, body));
```

Comment:
This needs to be updated to not use the hardcoded url but pushing now in order to show the redux state management is working first.